my-app/src/Pages/AllPosts/AllPosts.tsx

```diff
@@ -1,44 +1,103 @@
-import { useEffect, useState } from "react";
-import CardPostBig from "../../Components/CardPost/CardPostBig/CardPostBig";
+import React, { useEffect, useState } from "react";
 import CardPostMiddle from "../../Components/CardPost/CardPostMiddle/CardPostMiddle";
 import CardPostSmall from "../../Components/CardPost/CardPostSmall/CardPostSmall";
 import Title from "../../Components/Title/Title";
 import style from './AllPosts.module.scss'
 import Tabs from "../../Components/Tabs/Tabs";
 import {ReactComponent as Prev} from "../../assets/prev.svg";
 import {ReactComponent as Next} from "../../assets/next.svg";
-
-
+import { useDispatch, useSelector } from "react-redux";
+import { fetchPosts, setOrdering, setPage, setSearchQuery } from "../../store/postsSlice";
+import { useNavigate } from "react-router-dom";
 interface IPost {
     id: number;
     image?: string;
     text?: string;
     date: string;
     title: string;
+    index: number;
 }
 const AllPosts = () => {
-    const [results, setResults] = useState([]);
+    const dispatch = useDispatch<any>();
+    const navigate = useNavigate();
+    const {posts, loading, error, currentPage, itemsPerPage, totalItems, searchQuery, ordering} = useSelector((state) => state.posts);
     useEffect (() => {
-        fetch('https://studapi.teachmeskills.by/blog/posts/?author__course_group=14&format=json&limit=9')
-        .then((response) => response.json())
-        .then((json) => setResults((json.results)))
-    },[])
+        dispatch(fetchPosts({
+            limit: itemsPerPage,
+            offset: (currentPage - 1) * itemsPerPage,
+            searchQuery: searchQuery,
+            ordering: ordering,
+        }))
+    },[currentPage, ordering]);
+    if(loading) {
+        return <div>Loading...</div>;
+    }
+    if (error) {
+        return <div>Error</div>
+    }
+    const handlerPageChange = (pageNumber: number) => {
+        dispatch(setPage(pageNumber))
+    }
+    const handlerSubmit = (e:React.ChangeEvent<HTMLFormElement>) => {
+        e.preventDefault();
+        dispatch(fetchPosts>({
+            limit: itemsPerPage,
+            offset: 0,
+            searchQuery: searchQuery,
+            ordering: ordering,
+        }));
+        dispatch(setPage(1))
+    }
+    const handlerInput = (e:React.ChangeEvent<HTMLInputElement>) => {
+        const {value} = e.target
+        dispatch(setSearchQuery(value))
+    }
+    const handlerPrev = () => {
+        if(currentPage > 1)
+        dispatch(setPage(currentPage - 1))
+    }
+    const handlerNext = () => {
+        if(currentPage < totalPage)
+        dispatch(setPage(currentPage + 1))
+    }
+    const handlerOrdering = (e:React.ChangeEvent<HTMLSelectElement>) => {
+        dispatch(setOrdering(e.target.value))
+    }
+    const totalPage = Math.ceil(totalItems/itemsPerPage)
+    const renderPageNumber = () => {
```

Comment:
if you already know about useMemo or useCallbacks, it's better to wrap such functions in one of these hooks